vike/shared/route/loadPageRoutes.ts

```diff
@@ -175,12 +175,14 @@ function getGlobalHooks(
   pageConfigGlobal: PageConfigGlobalRuntime
 ): {
   onBeforeRouteHook: null | Hook
+  onBootHook: null | Hook
   filesystemRoots: null | FilesystemRoot[]
 } {
   // V1 Design
   if (pageConfigs.length > 0) {
-    const hook = getHookFromPageConfigGlobal(pageConfigGlobal, 'onBeforeRoute')
-    return { onBeforeRouteHook: hook, filesystemRoots: null }
+    const onBootHook = getHookFromPageConfigGlobal(pageConfigGlobal, 'onBoot')
+    const onBeforeRouteHook = getHookFromPageConfigGlobal(pageConfigGlobal, 'onBeforeRoute')
+    return { onBootHook, onBeforeRouteHook, filesystemRoots: null }
```

Comment:
It's a good question whether `onBoot()` should be global.

Maybe we can make `onBoot()` non-global and cumulative. In other words: the result of the various `onBoot()` hooks are cached and only the results of the relevant `onBoot()` hooks are applied each time Vike creates a new `pageContext` object.

~~But this means that `onBoot()` can be called later upon navigating to a new page and `onBoot()` is a misnomer then.~~ Ah, no, actually if make `onBoot()` eager then we can call *all* `onBoot()` hooks at boot time.

Reply:
I'm actually leaning more towards making `onBoot()` global as you suggested.

The thing is that, if we don't make it global, then we have to resolve routing before calling `onBoot()`, which is somewhat contradictory as an idea of `onBoot()` is to call it as early as possible and before any request is made to the server.

If a user wants a per-page hook, then he can use `onBeforeRender()`. (So far I don't think we need to introduce a second hook `onRequest()`.)

WDYT?